### ISSUE TITLE
Add `PrefixMap::children_mut` method

### ISSUE THREAD

**Issue by scotow** [tiborschneider/prefix-trie#12]
It would be nice to be able to modify children values when iterating over them. As of today, the workaround is to do this:

```rust
for (net, value) in map.children(&net).map(|(n, t)| (n, t.clone())).collect::<Vec<_>>() {
    value += 1;
    assert!(map.insert(net, value).is_some());
}
```

**Comment by tiborschneider** [comment]
Thanks for the comment. I don't even think the code you have above works as intended (it will not modify the fields in the map). But I agree that most of the iterators could (and maybe should) have a mutable variant as well (and perhaps even an owning variant that destructs the map directly). Sounds like a good use-case for macros. I will take a look at this.

**Comment by tiborschneider** [comment]
fixed in 8cd6bcde8614f17e50be91396553702a679e3133

**Comment by scotow** [comment]
Thanks for the quick implementation!
